build/ncollide2d/tests/geometry/ray_cast.rs

```diff
@@ -88,3 +88,54 @@ fn perpendicular_raycast_starting_bellow_segment() {
     let ray = Ray::new(Point2::new(0.0, -11.0), Vector2::new(1.0, 0.0));
     assert!(!segment.intersects_ray(&na::one(), &ray));
 }
+
+///    Ray Target
+///    +
+/// 3  |     Ray moved up each iteration
+/// |  v     ^
+/// 2  *--+  |
+/// |  |  |  +
+/// 1  +--+  *<----+Ray origin
+/// |
+/// 0--1--2--3
+///
+/// Tests the accuracy of raycaster collision detection against a `ConvexPolygon`.
+#[test]
+fn convexpoly_raycast_fuzz() {
+    let vertices: Vec<Point2<f64>> = vec![[1, 1], [1, 2], [2, 2], [2, 1]]
+        .into_iter()
+        .map(|[x, y]| Point2::new(x as f64, y as f64))
+        .collect();
+
+    let square = ConvexPolygon::try_new(vertices).unwrap();
+    let raycaster = square.as_ray_cast().unwrap();
+
+    let test_raycast = |ray_origin: Point2<f64>, ray_look_at: Point2<f64>| -> Option<f64> {
+        let mut ray_angle: Vector2<f64> = ray_look_at - ray_origin;
+        let ray_angle_magnitude =
+            ((ray_angle.x * ray_angle.x) + (ray_angle.y + ray_angle.y)).sqrt();
+        ray_angle /= ray_angle_magnitude;
```

Comment:
Line 115 to 117: there is already a feature on nalgebra to do this the normalization: `ray_angle.normalize()`. So you could just use: `Ray::new(ray_origin, ray_angle.normalize())` at line 121.